### PR TITLE
feature/APMI-2308-support NSURLConnection

### PR DESCRIPTION
**Pull request by shattimare**
verified commit

### CODE REVIEW

File: SplunkRumWorkspace/SplunkRum/SplunkRum/ConnectionInstrumentation.swift [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    (_ request: URLRequest?, returning response: AutoreleasingUnsafeMutablePointer<URLResponse?> )
+    throws -> Data {
+        let span = startConnectionSpan(request: request)
+        var data = Data()
+        do {
+            data = try splunk_swizzled_connection_sendSynchronousRequest(request, returning: response)
+            if let httpResponse = response.pointee as? HTTPURLResponse {
+                endConnectionSpan(connection: nil, hresponse: httpResponse, error: nil, span: span!)
+            }
+        } catch let error {
```
Comment (seemk) [inline SplunkRumWorkspace/SplunkRum/SplunkRum/ConnectionInstrumentation.swift:160]: Won't we break users' code if we don't rethrow?

File: SplunkRumWorkspace/SplunkRum/SplunkRum/ConnectionInstrumentation.swift [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import Foundation
+import OpenTelemetryApi
+// MARK: - NSURLConnection Instrumentation -
+/*class ConnectionObserver: NSObject {
```
Comment (seemk) [inline SplunkRumWorkspace/SplunkRum/SplunkRum/ConnectionInstrumentation.swift:20]: Do we need this large comment?

Comment (shattimare) [reply SplunkRumWorkspace/SplunkRum/SplunkRum/ConnectionInstrumentation.swift]: Done

File: SplunkRumWorkspace/SplunkRum/SplunkRum/ConnectionInstrumentation.swift (hunk 1 above)
Comment (shattimare) [reply SplunkRumWorkspace/SplunkRum/SplunkRum/ConnectionInstrumentation.swift:128]: throw error

**Pull request closed without merging**